docs/community/20241112-node-bootstrapping.md

```diff
@@ -0,0 +1,107 @@
+stakeholders for a Working Group dedicated to the Provisioning aspect of
+Node Bootstrapping.
+
+Provisioning, in this context, refers to configuring and customising the
+node operating system to prepare it to serve as a ClusterAPI cluster node.
+
+
+## User Story and Problem Statement
+
+As a Linux Distribution maintainer aiming to support ClusterAPI, or to improve
+my existing support of ClusterAPI in my distribution, I would like to be able to
+cleanly integrate my contributions into the node bootstrapping process without
+interfering with other implementations.
+I would like my contributions to be as re-usable as possible across different
+bootstrap providers.
+I would like to read and to follow documentation, guidelines, and specifications
+on the above.
+I would like to be able to do this to deliver choice, variety, and innovation to
+users of my own distribution as well as to the larger ClusterAPI ecosystem, and
+to enable other distribution vendors to do the same.
+
+
+**Problem statement / Example issues**
+
+Currently, the kubeadm bootstrap provider implements two provisioning systems:
+cloud-init and ignition.
+As there is currently no abstraction of node configuration, both implementations
+must cover generic configuration as well as specific config language concerns.
+This leads to duplicate code and to maintenance overhead.
+Also, other providers like e.g. MicroK8s must implement their own OS configuration
+generators since kubeadm's node provisioning implementations cannot be re-used.
+MicroK8s currently only supports cloud-init.
+
+Outside of ClusterAPI, both [cloud-init](https://github.com/canonical/cloud-init)
+and [ignition](https://github.com/coreos/ignition) provisioning is widely adopted
+across distributions.
+While cloud-init is used by general-purpose Linux distributions like Ubuntu/Debian,
+SUSE Linux, Alma, Rocky, Fedora, and Red Hat Enterprise Linux, ignition is popular
+with special-purpose distributions like Fedora CoreOS / Red Hat CoreOS, SuSE MicroOS
+/ SLE Micro / Kalpa, and Flatcar Container Linux.
+It is likely that more provisioning systems exist; a clean separation and guidelines
+will make it easier to add support to ClusterAPI as well as to share implementations
+across bootstrap providers.
+
+
+**Compatibility and API guarantees**
+
+ClusterAPI has been stable for multiple years and is in widespread production use.
+Proposals and implementations of this working group must ensure existing integrations
+continue to work.
+Any changes in the kubeadm bootstrap provider in particular must uphold guarantees and
+expectations set by the current implementation, and must continue to support current use
+cases.
+
+## Scope
+
+1. Propose architectural improvements that ease integration of node configuration
+     implementations.
+   Decouple generic bootstrapping from configuration language specific concerns.
+2. Deliver an example implementation in the kubeadm bootstrapper.
```

Comment:
```suggestion
2. Deliver an example implementation in the kubeadm bootstrap provider.
```

Better to use the [official terminology](https://cluster-api.sigs.k8s.io/reference/glossary#bootstrap-provider) to avoid confusion. Ditto all occurrences below.